### PR TITLE
add : `get_block_hash`, `get_block_by_num`, `get_signed_block_by_num`, `ApiClientError::Custom(dyn Box<std::Error>`

### LINKED ISSUE #123: add ApiClientError::Custom(String/ dyn Box)
This gives an end user more freedom without necessarily defining another error.

### PR DESCRIPTION
**Pull request by clangenb**
* Also closes #123

### CODE REVIEW

File: src/std/mod.rs [hunk 1]
```diff
@@ -246,13 +246,28 @@ where
...
         B: Block + DeserializeOwned,
     {
         Self::get_signed_block(self, hash).map(|sb_opt| sb_opt.map(|sb| sb.block))
     }
 
+    pub fn get_block_by_num<B>(&self, hash: Option<u32>) -> ApiResult<Option<B>>
+    where
+        B: Block + DeserializeOwned,
+    {
+        Self::get_signed_block_by_num(self, hash).map(|sb_opt| sb_opt.map(|sb| sb.block))
```
Comment (haerdib) [inline src/std/mod.rs:268]:
```suggestion
        Self::get_signed_block_by_num(self, number).map(|sb_opt| sb_opt.map(|sb| sb.block))
```

**Review by haerdib — CHANGES_REQUESTED** [review]
Minor change request

File: src/std/mod.rs [hunk 2]
```diff
@@ -246,13 +246,28 @@ where
...
+    }
+
     pub fn get_block<B>(&self, hash: Option<Hash>) -> ApiResult<Option<B>>
     where
         B: Block + DeserializeOwned,
     {
         Self::get_signed_block(self, hash).map(|sb_opt| sb_opt.map(|sb| sb.block))
     }
 
+    pub fn get_block_by_num<B>(&self, hash: Option<u32>) -> ApiResult<Option<B>>
```
[the remark this answers is not in the document]
Comment (clangenb) [reply src/std/mod.rs:264]: Oh, good catch

**Comment by clangenb** [comment]
Whatever you wish, mylady. :)

**Pull request merged**